src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -67,26 +62,25 @@ public boolean isProductInBasket(String name) {
 
    public void deleteBasket() {
          basketAll.clear();
-         count = 0;
+
    }
 
-   public List deleteOneTypeOfProduct(Product product) {
-      int howManyDeleted = 0;
-      List<Product> deletedProducts = new LinkedList<>();
-      if (count == 0) {
+   public LinkedList deleteOneTypeOfProduct(Product product) {
+
+      LinkedList<Product> deletedProducts = new LinkedList<>();
```

Comment:
Нужно всегда оперировать интерфейсными ссылками, это тема полиморфизма
```suggestion
      List<Product> deletedProducts = new LinkedList<>();
```